mws/utils/deprecation.py

```diff
@@ -7,3 +9,30 @@ class RemovedInPAM10Warning(DeprecationWarning):
 
 class RemovedInPAM11Warning(PendingDeprecationWarning):
     pass
+
+
+def kwargs_renamed_for_v11(kwarg_map):
+    """Warns of old kwargs from pre-1.0 being used. Notifies the user to call
+    the method using new kwargs, and that the old will be removed in v1.1.
+    """
+
+    def _decorator(request_func):
+        @wraps(request_func)
+        def _wrapped_func(self, *args, **kwargs):
+            for old, new in kwarg_map:
+                if old in kwargs:
+                    # An old argument was found; replace it with the new one, warn
+                    warnings.warn(
```

Comment:
Just a a quick test (as I've not played with the warnings module before)


```python
import warnings
print("hello")
warnings.warn("this is a warning", DeprecationWarning)
print("after warning")
```

When run does not raise any warnings in my console.

but
```python
import warnings
print("hello")
warnings.warn("this is a warning")
print("after warning")
```
does.

As you have subclassed the `DeprecationWarning` what are we expecting to happen here (or have I ballsed up my example?)

Reply:
Deprecation warnings are not typically surfaced to end users without changing warning filters.

In our case, warnings are actually forced to the surface in `mws/__init__.py`:

```python
warnings.simplefilter("default")
```

If you add that line to your quick test, you should see the deprecation warning pop up.

Reply:
👍 

Lovely - a new module for me to play with now!